hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sPipelineElementApi.java

```diff
@@ -0,0 +1,54 @@
+package com.linkedin.hoptimator.k8s;
+
+import com.linkedin.hoptimator.k8s.models.V1alpha1Pipeline;
+import com.linkedin.hoptimator.k8s.models.V1alpha1PipelineList;
+import com.linkedin.hoptimator.k8s.status.K8sPipelineElementStatus;
+import com.linkedin.hoptimator.k8s.status.K8sPipelineElementStatusEstimator;
+import com.linkedin.hoptimator.util.Api;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/** Provides all pipeline elements in a {@link com.linkedin.hoptimator.k8s.K8sContext} instance. */
+public class K8sPipelineElementApi implements Api<K8sPipelineElement> {
+  private final K8sContext context;
+
+  public K8sPipelineElementApi(K8sContext context) {
+    this.context = context;
+  }
+
+  private Collection<K8sPipelineElement> discoverAllElements(K8sContext context) {
+    final K8sApi<V1alpha1Pipeline, V1alpha1PipelineList> pipelineApi = new K8sApi<>(context, K8sApiEndpoints.PIPELINES);
+    Collection<V1alpha1Pipeline> pipelines;
+    try {
+      pipelines = pipelineApi.list();
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+
+    Map<String, K8sPipelineElement> elements = new HashMap<>();
+    K8sPipelineElementStatusEstimator statusEstimator = new K8sPipelineElementStatusEstimator(context);
+    for (V1alpha1Pipeline pipeline : pipelines) {
+      List<K8sPipelineElementStatus> elementStatuses = statusEstimator.estimateStatuses(pipeline);
+      for (K8sPipelineElementStatus elementStatus : elementStatuses) {
+        String key = elementStatus.getName();
```

Comment:
The element's name is not necessarily a unique key. For example, you can have a `Pod` named `foo` and a `KafkaTopic` named `foo`, but those are separate things.

Since this is inside the `k8s` module, we can assume that these objects have all have a `kind`, at least. I think the key should at least be a tuple of name and kind.

Theoretically, you could also use the `uid` of each object, which is globally unique.

Reply:
Ah, I see from the integration tests that `getName()` is actually returning `Kind/name`. Works for me.